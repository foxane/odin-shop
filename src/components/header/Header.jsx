import PropTypes from 'prop-types';
import { Link, NavLink } from 'react-router-dom';
import { useState } from 'react';
import useScroll from '../../utils/useScroll';
import styles from './Header.module.css';
import logo from '/logo.svg';

// cart count is not implemented yet, change state to 0!
const Header = ({ itemCount = 10 }) => {
  const [isNavOpen, setIsNavOpen] = useState(false);
  const isScrolled = useScroll();

  return (
    <header
      className={`${styles.container} ${isScrolled ? styles.scrolled : ''}`}
    >
      <div className={styles.contentWrapper}>
        <button
          aria-label="navigation button"
          className={`${styles.navBtn} ${styles.openMenuBtn}`}
          onClick={() => setIsNavOpen(!isNavOpen)}
        >
          <i className="fa-solid fa-bars"></i>
        </button>
        <Link to={'/'} className={styles.title}>
          <h1 className={styles.titleBig}>Gogobebe</h1>
          <img src={logo} alt="Gogobebe logo" className={styles.titleSmall} />
        </Link>
        <div className={styles.inlineNav}>
          <nav>
            <NavLink to={'/'}>Home</NavLink>
            <NavLink to={'cat'}>Categories</NavLink>
            <NavLink to={'products'}>All Products</NavLink>
            <NavLink to={'about'}>About</NavLink>
          </nav>
        </div>
        <div className={styles.btnContainer}>
          <button className={styles.navBtn}>
            <i className="fa-solid fa-magnifying-glass"></i>
          </button>
          <div className={styles.cartContainer}>
            <Link to="cart" className={styles.navBtn}>
              <i className="fa-solid fa-cart-shopping"></i>
              <span aria-label="item in cart" className={styles.itemCount}>
                {itemCount}
              </span>
            </Link>
          </div>
        </div>

        {/* Navigation panel */}
        {isNavOpen && (
          <>
            <div
              aria-label="navigation content"
              className={`${styles.floatNav} ${
                isNavOpen ? styles.open : styles.close
              }`}
            >
              <button
                onClick={() => setIsNavOpen(!isNavOpen)}
                className={`${styles.navBtn} ${styles.closeNavBtn}`}
              >
                <i className="fa-solid fa-xmark"></i>
              </button>
              <nav>
                <Link to={'/'}>Home</Link>
                <Link to={'cat'}>Categories</Link>
                <Link to={'products'}>All Products</Link>
                <Link to={'about'}>About</Link>
              </nav>
              <a
                href="http://github.com/foxane/odin-shop"
                target="_blank"
                rel="noopener noreferrer"
                className={styles.source}
              >
                <i className="fa-brands fa-github"></i> Source
              </a>
              <div
                className={styles.overlay}
                onClick={() => setIsNavOpen(!isNavOpen)}
              ></div>
            </div>
          </>
        )}
      </div>
    </header>
  );
};

export default Header;

Header.propTypes = {
  itemCount: PropTypes.number,
  errorPage: PropTypes.element,
};
