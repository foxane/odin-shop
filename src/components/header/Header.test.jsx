import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import Header from './Header';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

describe('Header component', () => {
  it('rendered correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  it('open and closing navbar', async () => {
    render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );
    const navBtn = screen.getByLabelText('navigation button');
    await userEvent.click(navBtn);
    // Use find instead of get because there's animation
    const navContent = await screen.findByLabelText('navigation content');
    expect(navContent).toBeInTheDocument();

    await userEvent.click(navBtn);
    expect(navContent).not.toBeInTheDocument();
  });

  it('show cart item count correctly', () => {
    const mockCount = 10;

    render(
      <MemoryRouter>
        <Header itemCount={mockCount} />
      </MemoryRouter>,
    );

    const itemCountEl = screen.getByLabelText('item in cart');
    expect(itemCountEl).toHaveTextContent(mockCount.toString());
  });
});
