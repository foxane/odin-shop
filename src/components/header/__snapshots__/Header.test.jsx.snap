// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > rendered correctly 1`] = `
<div>
  <header
    class="_container_69e643 "
  >
    <div
      class="_contentWrapper_69e643"
    >
      <button
        aria-label="navigation button"
        class="_navBtn_69e643 _openMenuBtn_69e643"
      >
        <i
          class="fa-solid fa-bars"
        />
      </button>
      <a
        class="_title_69e643"
        href="/"
      >
        <h1
          class="_titleBig_69e643"
        >
          Gogobebe
        </h1>
        <img
          alt="Gogobebe logo"
          class="_titleSmall_69e643"
          src="/logo.svg"
        />
      </a>
      <div
        class="_inlineNav_69e643"
      >
        <nav>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Home
          </a>
          <a
            class=""
            href="/categories"
          >
            Categories
          </a>
          <a
            class=""
            href="/products"
          >
            All Products
          </a>
          <a
            class=""
            href="/about"
          >
            About
          </a>
        </nav>
      </div>
      <div
        class="_btnContainer_69e643"
      >
        <button
          class="_navBtn_69e643"
        >
          <i
            class="fa-solid fa-magnifying-glass"
          />
        </button>
        <div
          class="_cartContainer_69e643"
        >
          <a
            class="_navBtn_69e643"
            href="/cart"
          >
            <i
              class="fa-solid fa-cart-shopping"
            />
            <span
              aria-label="item in cart"
              class="_itemCount_69e643"
            >
              10
            </span>
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
